src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -48,26 +50,51 @@ class IpexCommunicationService extends AgentService {
   protected readonly notificationStorage: NotificationStorage;
   protected readonly ipexMessageStorage: IpexMessageStorage;
   protected readonly operationPendingStorage: OperationPendingStorage;
+  protected readonly multisigService: MultiSigService;
+  protected readonly signifyNotificationService: SignifyNotificationService;
 
   constructor(
     agentServiceProps: AgentServicesProps,
     identifierStorage: IdentifierStorage,
     credentialStorage: CredentialStorage,
     notificationStorage: NotificationStorage,
     ipexMessageStorage: IpexMessageStorage,
-    operationPendingStorage: OperationPendingStorage
+    operationPendingStorage: OperationPendingStorage,
+    multisigService: MultiSigService,
+    signifyNotificationService: SignifyNotificationService
   ) {
     super(agentServiceProps);
     this.identifierStorage = identifierStorage;
     this.credentialStorage = credentialStorage;
     this.notificationStorage = notificationStorage;
     this.ipexMessageStorage = ipexMessageStorage;
     this.operationPendingStorage = operationPendingStorage;
+    this.multisigService = multisigService;
+    this.signifyNotificationService = signifyNotificationService;
   }
 
   @OnlineOnly
   async acceptAcdc(id: string): Promise<void> {
-    const grantNoteRecord = await this.getNotificationRecordById(id);
+    const grantNoteRecord = await this.notificationStorage.findById(id);
+
+    if (!grantNoteRecord) {
+      throw new Error(
+        `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
+      );
+    }
+
+    if (
+      grantNoteRecord.linkedGroupRequests &&
+      Object.keys(grantNoteRecord.linkedGroupRequests).length
+    ) {
+      for (const said of Object.keys(grantNoteRecord.linkedGroupRequests)) {
+        if (!grantNoteRecord.linkedGroupRequests[said]) {
+          await this.acceptAcdcFromMultisigExn(said as string);
```

Comment:
This should only be called from here - right now it's still called from `ReceiveCredential.tsx`. Can you tidy up that check to only call this function?